fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnector.java

```diff
@@ -3,58 +3,61 @@
 package com.starrocks.connector.hive;
 
 import com.google.common.base.Preconditions;
-import com.starrocks.common.Config;
 import com.starrocks.common.util.Util;
 import com.starrocks.connector.Connector;
 import com.starrocks.connector.ConnectorContext;
 import com.starrocks.connector.ConnectorMetadata;
-import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.external.RemoteFileIO;
+import com.starrocks.external.hive.IHiveMetastore;
 
 import java.util.Map;
 
 public class HiveConnector implements Connector {
     public static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
     private final Map<String, String> properties;
-    private String resourceName;
-    private ConnectorMetadata metadata;
+    private final String catalogName;
+    private final HiveConnectorInternalMgr internalMgr;
+    private final HiveMetadataFactory metadataFactory;
 
     public HiveConnector(ConnectorContext context) {
         this.properties = context.getProperties();
+        this.catalogName = context.getCatalogName();
+        this.internalMgr = new HiveConnectorInternalMgr(catalogName, properties);
+        this.metadataFactory = createMetadataFactory();
         validate();
         onCreate();
     }
 
     public void validate() {
-        this.resourceName = Preconditions.checkNotNull(properties.get(HIVE_METASTORE_URIS),
+        String hiveMetastoreUris = Preconditions.checkNotNull(properties.get(HIVE_METASTORE_URIS),
                 "%s must be set in properties when creating hive catalog", HIVE_METASTORE_URIS);
-        Util.validateMetastoreUris(resourceName);
+        Util.validateMetastoreUris(hiveMetastoreUris);
     }
 
     @Override
     public ConnectorMetadata getMetadata() {
-        if (metadata == null) {
-            metadata = new HiveMetadata(resourceName);
-        }
-        return metadata;
+        return metadataFactory.create();
     }
 
+    private HiveMetadataFactory createMetadataFactory() {
+        IHiveMetastore metastore = internalMgr.createHiveMetastore();
+        RemoteFileIO remoteFileIO = internalMgr.createRemoteFileIO();
+        return new HiveMetadataFactory(
+                catalogName,
+                metastore,
+                remoteFileIO,
+                internalMgr.getHiveMetastoreConf(),
+                internalMgr.getRemoteFileConf(),
+                internalMgr.getPullRemoteFileExecutor(),
+                internalMgr.isSearchRecursive()
+        );
+    }
 
     public void onCreate() {
```

Comment:
why not remove this function?

Reply:
After we have adapted auto sync metadata, we still need this operation. Daily test doesn't include auto sync's case. so I just temporarily delete this part of code